### PR TITLE
Implement logic to replicate user data update event notification to queue

### PR DESCRIPTION
**Pull request by KevinSi96**
#### List of Changes

When user event type is UPDATE, the microservice will send a delete event and active event 

#### Motivation and Context

FD reads user data changes as a delete and activation events

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: connector/kafka-manager/src/main/java/it/pagopa/selfcare/external_interceptor/connector/kafka_manager/factory/SendFdNotification.java [hunk 1]
```diff
@@ -55,18 +60,51 @@ public void sendUserNotification(UserNotification userNotification, Acknowledgme
...
+                if(userProduct.getOnboardedProductDetails() != null) {
+                    sendUpdateUserEvents(notificationToSend, userProduct);
+                    acknowledgment.acknowledge();
+                }
+            } else {
+                notificationToSend.setType(NotificationType.getNotificationTypeFromRelationshipState(userNotification.getUser().getRelationshipStatus()));
+                String userNotificationToSend = mapper.writeValueAsString(notificationToSend);
+                String topic = producerAllowedTopics.get().get(userNotification.getProductId());
+                String logSuccess = String.format("sent notification for user : %s, to FD", userNotification.getUser().getUserId());
+                String logFailure = String.format("error during notification sending for user %s: {}, on FD ", userNotification.getUser().getUserId());
```
Comment (manuraf) [inline connector/kafka-manager/src/main/java/it/pagopa/selfcare/external_interceptor/connector/kafka_manager/factory/SendFdNotification.java:74]: these lines are duplicated....you could move them to sendNotification method

File: connector/kafka-manager/src/main/java/it/pagopa/selfcare/external_interceptor/connector/kafka_manager/factory/SendFdNotification.java [hunk 2]
```diff
@@ -55,18 +60,51 @@ public void sendUserNotification(UserNotification userNotification, Acknowledgme
...
+            notification.setUser(userToSend);
+            String userNotificationToSend = null;
+            try {
+                userNotificationToSend = mapper.writeValueAsString(notification);
+            } catch (JsonProcessingException e) {
+                throw new RuntimeException(e);
+            }
+            String topic = producerAllowedTopics.get().get(notification.getProduct());
+            String logSuccess = String.format("sent notification for user : %s, to FD", notification.getUser().getUserId());
+            String logFailure = String.format("error during notification sending for user %s: {}, on FD ", notification.getUser().getUserId());
```
Comment (manuraf) [inline connector/kafka-manager/src/main/java/it/pagopa/selfcare/external_interceptor/connector/kafka_manager/factory/SendFdNotification.java:101]: you could introduce a constant class variabile for the string message, for ex. MESSAGE_LOG_SUCCESS

File: connector/rest/pom.xml [hunk 3]
```diff
@@ -15,6 +15,11 @@
             <groupId>it.pagopa.selfcare</groupId>
             <artifactId>selc-commons-connector-rest</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.cloud</groupId>
+            <artifactId>spring-cloud-starter-openfeign</artifactId>
+            <version>4.0.4</version>
```
Comment (manuraf) [inline connector/rest/pom.xml:21]: version should be retrieved automatically by spring, isn't?

File: connector/rest/src/main/java/it/pagopa/selfcare/external_interceptor/connector/rest/client/ExternalApiBackEndRestClient.java
Comment (manuraf) [inline connector/rest/src/main/java/it/pagopa/selfcare/external_interceptor/connector/rest/client/ExternalApiBackEndRestClient.java:1]: you might use openapi-generator-maven-plugin for generating rest client

**Review by manuraf — CHANGES_REQUESTED** [review]

File: connector/kafka-manager/src/main/java/it/pagopa/selfcare/external_interceptor/connector/kafka_manager/factory/SendFdNotification.java (hunk 1 above)
Comment (KevinSi96) [reply connector/kafka-manager/src/main/java/it/pagopa/selfcare/external_interceptor/connector/kafka_manager/factory/SendFdNotification.java:74]: the log message for FD and SAP have different formats, I will indroduce a CONSTANT for them

**Review by manuraf — APPROVED** [review]

**Review by gianmarcoplutino — APPROVED** [review]

**Pull request merged**